### PR TITLE
A new implementation of the same ideas using Cassette!

### PR DESCRIPTION
**Pull request by NHDaly**
I think I got a bit carried away with the parse-time stuff, and i should rely
more on Cassette, but it's a hard habit to break. I'll need to look at this
more again. :)

### CODE REVIEW

File: ide/cassette.jl [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+# -------- whole file
+
+include("parsefile.jl")
+ctx = LiveCtx(metadata = CollectedOutputs([], 0))
+@time @eval Cassette.overdub(ctx,
+    ()->$(thunkwrap_toplevel(parseall(read("$(@__DIR__)/../examples/example1.jl", String)))))
+@show ctx.metadata.outputs
+
+
+# -----------
```
Comment (rai-nhdaly) [inline ide/cassette.jl]:
```suggestion
# -------------
```

File: ide/cassette.jl [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    foo(3)
+    M.f("hey")
+end
+end)))
+@show ctx.metadata.outputs
+foo
+
+Core.eval(Main, quote Core.eval(Main, (()->quote f2(x) = x+1 end)()) end)
+
+# -------- whole file
```
Comment (rai-nhdaly) [inline ide/cassette.jl]:
```suggestion
# ---------- whole file
```

**Comment by NHDaly** [comment]
It's not quite right yet, since it's printing duplicate values inside functions, and it's still just as slow as the old version, but it's neat that we're replacing ~450 lines with ~150 lines.


screen shot 2019-01-22 at 5 35 48 pm

And I _think_ it's a good thing that you don't have to decorate the functions with the special `Live.@test a=1 b=3` syntax. I think just intercepting function calls through cassette is way cooler!

**Comment by NHDaly** [comment]
Okay, fixed the double-printing by making the "last line number" tracker a stack instead of a global.

I could maybe also fix this by defining `Cassette.execute(::LiveCtx, ::Thunk)` instead of `posthook`, but `execute` is confusing. I tried it and ran into world-age errors (probably due to my liberal use of `@eval`.


Sadly though, the stack made it a good deal slower.

**Comment by NHDaly** [comment]
screen shot 2019-01-22 at 6 40 06 pm

**Pull request merged**